hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/translators/ConsensusSubmitMessageTranslator.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.junit.support.translators;
+
+import com.hedera.hapi.block.stream.output.StateChange;
+import com.hedera.hapi.block.stream.output.StateChanges;
+import com.hedera.hapi.node.transaction.TransactionReceipt;
+import com.hedera.hapi.node.transaction.TransactionRecord;
+import com.hedera.node.app.state.SingleTransactionRecord;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+public class ConsensusSubmitMessageTranslator implements TransactionRecordTranslator<SingleTransactionBlockItems> {
+    @Override
+    public SingleTransactionRecord translate(
+            @NotNull SingleTransactionBlockItems transaction, @Nullable StateChanges stateChanges) {
+        final var receiptBuilder = TransactionReceipt.newBuilder();
+
+        // TODO: Where in state is the topicRunningHashVersion stored? Hardcode to 3 for now.
+        receiptBuilder.topicRunningHashVersion(3L);
```

Comment:
I think this will come from the state changes. I'll add an example of what I _think_ needs to go here

Reply:
I was wrong, the `Topic` object doesn't have the running hash _version_ at all, so the state changes won't help :( we're going to need to ask @jsync-swirlds why this property was removed (or never present) in a consensus submit's `TransactionOutput`

Reply:
The `version` of the topic running hashes never changes, so it was removed from the output as unnecessary repetition.
It's version 3 back to some quite early record streams and very unlikely to change, so it's not included in the block stream submit output because that wastes 3-5 bytes for each HCS submit, which is the most frequent transaction we have.

The assumption is that to change the running hash we have to deploy a new consensus node software version, and run a HIP for the change, so if we ever do make that change, we'll add the version as a separate state singleton (so we only transmit the version when it actually changes).
If the mirror node needs that value repeated for every HCS submit, it can be hardcoded when reading block stream input.

We also have an express assumption that block stream _does not_ need to be convertible to a record stream; the two are _not_ compatible.  There _must_ be at *least* enough data in block stream to support mirror node, but it need not be in a form that makes recreating the record stream reasonable, or even possible.

Reply:
Ok, makes sense. If it's not already, we should document this in the protobufs. 

The reason we're working on converting block streams to records is for diff testing: compare current outputs to the new block stream outputs and make sure they are as expected

Reply:
It is documented in the block stream protobuf (one of the many items that are pending a HIP for a major "specification" update).
The purpose here is understood, and reasonable; it just might require a bit more contortions than might be preferred 🤷.

Reply:
> It is documented in the block stream protobuf
Awesome, you're on top of it :)

> [...]so even perfect match in diff testing won't guarantee a fully correct block stream (though it will certainly help)
Agreed, and we won't go crazy making them identical. Thanks!

Reply:
Remove the `TODO` and went ahead with using the constant value for running hash version.